transport/src/main/java/io/netty/channel/DefaultMaxMessagesRecvByteBufAllocator.java

```diff
@@ -48,6 +49,38 @@ public MaxMessagesRecvByteBufAllocator maxMessagesPerRead(int maxMessagesPerRead
         return this;
     }
 
+    /**
+     * Determine if future instances of {@link #newHandle()} will stop reading if we think there is no more data.
+     * @param respectMaybeMoreData
+     * <ul>
+     *     <li>{@code true} to stop reading if we think there is no more data. This may save a system call to read from
+     *          the socket, but if data has arrived in a racy fashion we may give up our {@link #maxMessagesPerRead()}
+     *          quantum and have to wait for the selector to notify us of more data.</li>
+     *     <li>{@code false} to keep reading (up to {@link #maxMessagesPerRead()}) or until there is no data when we
+     *          attempt to read.</li>
+     * </ul>
+     * @return {@code this}.
+     */
+    public MaxMessagesRecvByteBufAllocator respectMaybeMoreData(boolean respectMaybeMoreData) {
```

Comment:
Should we change the return type to `DefaultMaxMessagesRecvByteBufAllocator`? Also should we override this methods in the other impls and return the correct type there as well ?

Reply:
done